paho/pinger.go

```diff
@@ -109,13 +111,20 @@ func (p *DefaultPinger) Run(ctx context.Context, conn net.Conn, keepAlive uint16
 				timer.Reset(pingDue.Sub(t))
 				continue
 			}
-
-			lastPingSent = time.Now() // set before sending because WriteTo may return after PINGRESP is handled
-			if _, err := packets.NewControlPacket(packets.PINGREQ).WriteTo(conn); err != nil {
-				p.debug.Printf("DefaultPinger packet write error: %v", err)
-				return fmt.Errorf("failed to send PINGREQ: %w", err)
-			}
+			lastPingSent = time.Now()
+			go func() {
+				if _, err := packets.NewControlPacket(packets.PINGREQ).WriteTo(conn); err != nil {
+					p.debug.Printf("DefaultPinger packet write error: %v", err)
+					errCh <- fmt.Errorf("failed to send PINGREQ: %w", err)
+					return
+				}
+				errCh <- nil
+			}()
 			timer.Reset(interval)
+		case err := <-errCh:
+			if err != nil {
+				return err
```

Comment:
Probably worth stopping the timer here too (in the previous version the `return` occurred before the timer had fired so this was not necessary). This is a minor point as the timer would fire eventually but think it's tidier (and there could be an issue in a connect/disconnect loop with a long keepalive period).

Reply:
Fixed and added a comment. As I understood it, the concern is that it's a potential temporary "memory leak"

Reply:
More that it was creating unnecessary activity (sending a `nil` on the channel and then always ignoring the `nil`). I also did suspect that there was a possible route through the code where a go-routine could have been leaked (but did not really follow through investigating this as it's simpler just to remove the unnecessary send).